template/runner/base/base_setup.py

```diff
@@ -644,6 +647,9 @@ def checkpoint(cls, epoch, new_value, best_value, log_dir, the_lower_the_better=
         # If enabled, save all checkpoints with epoch number.
         if checkpoint_all_epochs:
             shutil.move(filename, os.path.join(log_dir, f'checkpoint_{epoch}.pth.tar'))
+        if checkpoint_every is not None:
+            if (epoch+1) % checkpoint_every == 0:
+                shutil.copyfile(filename, os.path.join(log_dir, f'checkpoint_{epoch}.pth.tar'))
```

Comment:
is there a reason to use copyfile while two rows above we use move? also, what happens if checkpoint_all_epochs moves it and then you try to copy it? does it crash ?

Reply:
- While move is probably a bit faster, copyfile allows the checkpoint to stay in place. So if you want to just pick up the very last checkpoint that was saved, it makes it easier. 

- Good point, i'll switch the condition to be `elif checkpoint_every is not None`